OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -712,13 +712,17 @@ void QueueCaptureOrders()
 
 			var capturableTargetOptions = targetOptions
 				.Select(a => new CaptureTarget<CapturableInfo>(a, "CaptureActor"))
-				.Where(target => target.Info != null && capturers.Any(capturer => target.Info.CanBeTargetedBy(capturer, target.Actor.Owner)))
+				.Where(target => target.Info != null && capturers.Any(capturer =>
+					target.Actor.TraitOrDefault<Capturable>() != null &&
```

Comment:
This does `2 * capturers.Length` times more trait queries than it needs to.  You should be able to reduce it to one query per actor by doing something like:
```csharp
Func<Actor, Actor[], bool> canCapture = (targetActor, candidateCaptors) =>
{
	var capturable = targetActor.TraitOrDefault<Capturable>();
	if (capturable == null)
		return false;

	return candidateCaptors.Any(c => capturable.CanBeTargetedBy(c, targetActor.Owner));
};

var capturableTargetOptions = targetOptions
	.Select(a => new CaptureTarget<CapturableInfo>(a, "CaptureActor"))
	.Where(target => target.Info != null && canCapture(target.Actor, capturers))
	.OrderByDescending(target => target.Actor.GetSellValue())
	.Take(maximumCaptureTargetOptions);
```